contracts/bridge/x/v4/RegistrationV4.sol

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Immutable Pty Ltd 2018 - 2024
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
+
+import {CoreV4} from "./CoreV4.sol";
+
+contract RegistrationV4 {
```

Comment:
Please add contract level documentation:
/** 
  * RegistrationV4 is a wrapper around the StarkEx contract to provide a more user-friendly interface for executing 
  *  multiple transactions on the StarkEx contract at once.
  *
 * This contract is not upgradeable. If an issue is found with this contract, a new version will be deployed.
 */